### DISCUSSION TITLE
Any way to pass a parameter to a window in avalonia?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15285] [Q&A]
I need to pass a parameter to a window in avalonia, but doing so causes the design view to break with the error:

System.Xaml.XamlException: Cannot compile Build method. Parameters mismatch. Type needs to have a parameterless ctor or a ctor with a single IServiceProvider argument. Or x:Arguments directive with matching arguments needs to be set.

The code like:

`public partial class TestWindow: Window
{
    public TestWindow(string mode="")
    {
         InitializeComponent();
    }
}`

**Comment by timunie** [comment]
just add another parameterless constuctor: 

```cs
public TestWindow(string mode="") : this()
{
    // do your stuff
}

public TestWindow() // maybe internal is enough, up to you to test
{ 
    InitializeComponent(); 
}

**Reply by maxkatz6** [reply]
Yes, otherwise there is no way for designer to create an instance of the type.
